osu.Game/Skinning/ResourcesSkin.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.OpenGL.Textures;
+using osu.Framework.Graphics.Textures;
+using osu.Framework.IO.Stores;
+using osu.Framework.Platform;
+using osu.Game.Audio;
+
+namespace osu.Game.Skinning
+{
+    /// <summary>
+    /// An <see cref="ISkin"/> that uses an underlying <see cref="IResourceStore{T}"/> with namespaces for resources retrieval.
+    /// </summary>
+    public class ResourcesSkin : ISkin
+    {
+        private readonly TextureStore textures;
+        private readonly ISampleStore samples;
+
+        public ResourcesSkin(IResourceStore<byte[]> resources, GameHost host, AudioManager audio)
+        {
+            textures = new TextureStore(host.CreateTextureLoaderStore(new NamespacedResourceStore<byte[]>(resources, @"Textures")));
+            samples = audio.GetSampleStore(new NamespacedResourceStore<byte[]>(resources, @"Samples"));
+        }
+
+        public Drawable? GetDrawableComponent(ISkinComponent component) => null;
+
+        public Texture? GetTexture(string componentName, WrapMode wrapModeS, WrapMode wrapModeT) => textures.Get(componentName, wrapModeS, wrapModeT);
+
+        public ISample? GetSample(ISampleInfo sampleInfo)
+        {
+            foreach (var lookup in sampleInfo.LookupNames)
+            {
+                ISample? sample = samples.Get(lookup);
+                if (sample != null)
+                    return sample;
+            }
+
+            return null;
+        }
+
+        public IBindable<TValue>? GetConfig<TLookup, TValue>(TLookup lookup) => null;
+
+        #region Disposal
+
+        ~ResourcesSkin()
+        {
+            // required to potentially clean up sample store from audio hierarchy.
+            Dispose(false);
+        }
```

Comment:
This should not be a thing. Explicitly dispose this "ResourcesSkin" from the place it's constructed instead.

Reply:
Yeah was wondering about this when you mentioned that finalizers are banned. I guess the rule should be that if the object can be explicitly disposed (which should always be the case), then no finalizer should be added for it.